src/hotspot/share/jeandle/jeandleAbstractInterpreter.cpp

```diff
@@ -1036,12 +1039,66 @@ bool JeandleAbstractInterpreter::inline_intrinsic(const ciMethod* target) {
       _jvm->fpush(_ir_builder.CreateIntrinsic(JeandleType::java2llvm(BasicType::T_FLOAT, *_context), llvm::Intrinsic::fabs, {_jvm->fpop()}));
       break;
     }
+    case vmIntrinsicID::_dsin: {
+      if (StubRoutines::dsin() != nullptr) {
+        _jvm->dpush(call_vm((address) StubRoutines::dsin(), "StubRoutines::dsin", T_DOUBLE, {T_DOUBLE}, true));
+      } else {
+        _jvm->dpush(call_vm((address) SharedRuntime::dsin, "SharedRuntime::dsin", T_DOUBLE, {T_DOUBLE}, true));
+      }
+      break;
+    }
     default:
       return false;
   }
   return true;
 }
 
+// Generate IR for call into runtime stub
+llvm::CallInst* JeandleAbstractInterpreter::call_vm(address c_func, llvm::StringRef name, BasicType return_type, llvm::ArrayRef<BasicType> arg_types, bool is_leaf) {
```

Comment:
How about add Jeandle used hotspot routines to `JeandleRuntimeRoutine`, their address will be resolved by `JeandleCompiledCode`. Through this approach, we can uniformly manage all external routines of Jeandle, and in LLVM IR, each routine will be declared as a function, allowing us to determine which routine is being called by its function name.

Reply:
We can merge this pr first, then do this in another pr, this depends on you.

Reply:
Hotspot routines are wrapped as JeandleRuntimeRoutine. 

But I find I can not assign `gc-leaf-function` attribute to the call. It looks it is not a standard llvm attribute. We can resolve it in future PR.

Reply:
LLVM's attributes are designed to be very flexible - we just need to add this string, and it can be processed by other passes. If it cannot be recognized, there must be some other issues.
But we can surely make it in another pr, I will resolve this comment.

Reply:
A new issue is created to trace this https://github.com/jeandle/jeandle-jdk/pull/88#discussion_r2354477883